### PR TITLE
[MIG] event_email_reminder: Migrate to v12

### PR DESCRIPTION
**Pull request by yajo**
Standard migration.

@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
CIs red

**Comment by yajo** [comment]
Let's see now.

**Comment by pedrobaeza** [comment]
Still no luck

**Comment by yajo** [comment]
Oops, let's see now.

**Comment by yajo** [comment]
This time I think is the good one. :stuck_out_tongue:

File: event_email_reminder/models/event.py [hunk 1]
```diff
@@ -42,27 +41,26 @@ def run_event_email_reminder(
...
+        limit_date = today + timedelta(days=days)
         if draft_events:
             domain = [('state', 'in', ['draft', 'confirm'])]
         else:
             domain = [('state', '=', 'confirm')]
         if not near_events:
             domain.extend([
-                ('date_begin', '>=', '%s 00:00:00' % limit_date),
-                ('date_begin', '<=', '%s 23:59:59' % limit_date),
+                ('date_begin', '>=', datetime.combine(limit_date, time_start)),
```
Comment (pedrobaeza) [inline event_email_reminder/models/event.py]: AFAIK, domains still use string format, so I think this is working by chance as the `repr` of the datetime matches the format, but that can be assured. You should do `fields.Datetime.to_string(...`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please fix the cron data XML:
```
2019-04-23 11:52:09,263 160 WARNING openerp_test odoo.models: ir.cron.create() with unknown fields: args, function
```

File: event_email_reminder/models/event.py (hunk 1 above)
Comment (yajo) [reply event_email_reminder/models/event.py]: I was curious. Actually it's the other way around: [if you use a string, it's converted to a datetime object. If you use a datetime, it's used raw](https://github.com/odoo/odoo/blob/9617e79e583aeb3d6d983045c5934dae28dce0c1/odoo/osv/expression.py#L1101-L1115).

It's interesting, though, that I could have saved the ~~`00:00:00` and `23:59:59`~~ `datetime.combine()` thingies. :thinking:

**Comment by yajo** [comment]
Done.

File: event_email_reminder/models/event.py (hunk 1 above)
Comment (pedrobaeza) [reply event_email_reminder/models/event.py]: Well, seeing the code, it seems both are equally accepted, so good for that "polymorphism".

**Comment by pedrobaeza** [comment]
Now runbot and Travis are red.

**Comment by yajo** [comment]
Yikes, a typo.

**Review by pedrobaeza — APPROVED** [review]

**Review by victor-ditay — APPROVED** [review]

**Comment by yajo** [comment]
Is it ugly if I merge it myself? :thinking:

**Comment by pedrobaeza** [comment]
I save you the pain, hehe

**Pull request merged**